packages/playwright-core/src/server/injected/selectorUtils.ts

```diff
@@ -53,7 +53,7 @@ export function matchesAttributePart(value: any, attr: AttributeSelectorPart) {
 
 export function shouldSkipForTextMatching(element: Element | ShadowRoot) {
   const document = element.ownerDocument;
-  return element.nodeName === 'SCRIPT' || element.nodeName === 'NOSCRIPT' || element.nodeName === 'STYLE' || document.head && document.head.contains(element);
+  return element.nodeName === 'SCRIPT' || element.nodeName === 'STYLE' || document.head && document.head.contains(element);
```

Comment:
This goes against the idea of text selector - it only accounts for text that is for the end user. That's why it ignores things like `<script>` or `<noscript>`. I also think changing this today will break existing tests, so we should not do that.

If anything, we can (maybe??) do this when javascript is disabled. However, that's some extra plumbing.

---

I'd like to step back and figure out why would you want selecting by text in the `<noscript>` element? If you'd like to test that your page has proper `<noscript>` support, just do something like:

```ts
await expect(page.locator('noscript')).toHaveText('....');
```

This way you rely on the browser to handle `noscript` as it would. There is no need to test that browser shows `noscript` content when it should - that's better to leave for the browser developers.

Reply:
> This goes against the idea of text selector - it only accounts for text that is for the end user. That's why it ignores things like <script> or <noscript>.

This very simple "Please enable JS in your browser" messages displayed by nearly any JS metaframeworks is very simple example for a readable text message targeting end users!

> I also think changing this today will break existing tests, ...

The present behavior of playwright doesn't represent the actual behavior of real world browsers.
If we fix this flaw, it will definitely affect some (but in fact **very few**) existing tests.

But if we don't fix this incorrect behavior, we are not able to reliable tests those cases at all -- all those cases, where the mentioned changes really make a significant difference worth testing.

But to argue a little bit more realistic:  
Do you see any negative side effects or unforeseen consequences affecting any of the rather wide spectra of practial applications represented by playwrights own integration tests?

> I'd like to step back and figure out why would you want selecting by text in the <noscript> element?

I just want't to test, that the output seen by end users looks correct.

Using a text selection is the simplest form of a test to archive this goal. It should simple work as expected, because it would highly irritate all playwright users, if it doesn't work for this kind of visible output in browsers just like in all other cases.

And it's also important to fix this kind of selection facility, because otherwise you have to modify the output (add test IDs etc.) to locate the relevant spans. It therefore enables minimal intrusive kind of observation, which do not need any additional preparation or modification of the analyzed object.

> If you'd like to test that your page has proper <noscript> support, just do something like: ...

I don't think, your suggested alternative will work.

The `<noscript>` tag has rather uncommon semantic specified by the html standard.
If JS is enabled, you will not even find it in the DOM hierachie, but otherwise (=JS is present) it's simple showing all the included text or html code fragment in an utterly transparent manner.

Please simply take a look at the output of the included test case and the actual change in behavior of playwright caused by this fix. That's hopefully more convincing than theoretical debates.